snapcraft/internal/build_providers/_base_provider.py

```diff
@@ -90,5 +121,78 @@ def launch_instance(self) -> None:
     def setup_snapcraft(self) -> None:
         self.echoer.info('Setting up snapcraft in {!r}'.format(
             self.instance_name))
+        if common.is_snap():
+            self._inject_snapcraft()
```

Comment:
When snapcraft is running under `snap try`, this will be true, but then will `_inject_snapcraft` explode?

Reply:
Yes it will. I plan to solve that in a follow up PR, it is of low exposure as `snap try` of snapcraft is used by just you and me.